### ISSUE TITLE
Spring MVC model not forwarded

### ISSUE THREAD

**Issue by blaluc** [casid/jte#378]
It seems that while forwarding requests, the model instance is not injected from the request scope.

e.g.:
```
@Controller
@RequestMapping("/second")
public class OtherController {

	@GetMapping
	public String view2(Model model) {
		model.addAttribute("key2", "second value");
		return "forward:/";
	}
```
```
@Controller
@RequestMapping("/")
public class TestController {

	@GetMapping
	public String index(Model model) {
		model.addAttribute("key1", "first value");
		return "index";
	}
}
```

index.jte:
```
@param String key1
@param String key2

key1: ${key1} <br>
key2: ${key2}
```

By calling http://localhost:8080/second, the `key2` value should be propagated to the template and displayed, but it is not, and the following page is shown instead:
```
key1: first value
key2:
```

The same example with thymeleaf works as expected giving the result:
```
key1: first value
key2: second value
```

**Comment by blaluc** [comment]
it seems that can be solved by adding 
```
this.setExposeRequestAttributes(true);
```
in the `JteViewResolver` class, i.e.:

```
public class JteViewResolver extends AbstractTemplateViewResolver {
...
    public JteViewResolver(TemplateEngine templateEngine, final String templateSuffix) {
        this.templateEngine = templateEngine;
        this.setSuffix(templateSuffix);
        this.setViewClass(JteView.class);
        this.setContentType(MediaType.TEXT_HTML_VALUE);
        this.setOrder(Ordered.HIGHEST_PRECEDENCE);
        this.setExposeRequestAttributes(true);
    }
...
```

**Comment by casid** [comment]
@blaluc thanks for sharing. Your suggested fix looks good to me.

However I'm not a heavy spring user and don't know if this has any unintended side effects.

@atomfrede, @tschuehly would this change be okay with you?

**Comment by atomfrede** [comment]
At first glance it looks okay. Found similar issues with pebble for example. Not sure why it works for thymeleaf out of the box because it looks like the `exposeRequestAttributes` is not set (at least not explicitly).

**Comment by atomfrede** [comment]
I have found why it works with thymeleaf. In case it is forwarded, thymeleaf creates an `InternalViewResourceView` and in there it [exposes the model attributes as request attributes](https://github.com/spring-projects/spring-framework/blob/594ed95f3c868e5d2491645142dc787dbb2deebc/spring-webmvc/src/main/java/org/springframework/web/servlet/view/InternalResourceView.java#L142) (always). So setting it looks fine to me (maybe check if that has any implications on normal view resolving)

**Comment by atomfrede** [comment]
@blaluc Just curious, whats the use case for the forward (instead e.g. redirect with flash attributes for example), besides being the forward totally server side?

**Comment by blaluc** [comment]
Hi @atomfrede, the use case I’m working on involves integrating with the htmx library (https://htmx.org/). To handle a full-page reload, it might be useful to forward the model built in a 'partial' controller method (server-side) to another controller method responsible for rendering the full page.

**Comment by casid** [comment]
@blaluc did you check if it has any impact on normal view resolving? And, would you be up to submit a Pull Request with your suggested fix?

**Comment by blaluc** [comment]
Done (https://github.com/casid/jte/pull/383).

To avoid any impact I have added a configuration parameter (the default behavior is the previous one).

Cheers!

**Comment by casid** [comment]
This is now released with jte 3.1.13
